tests/components/hassio/test_binary_sensor.py

```diff
@@ -0,0 +1,159 @@
+                "version": "1.0.0",
+                "update_available": False,
+            },
+        },
+    )
+    aioclient_mock.get(
+        "http://127.0.0.1/supervisor/info",
+        json={
+            "result": "ok",
+            "data": {
+                "result": "ok",
+                "version_latest": "1.0.0",
+                "addons": [
+                    {
+                        "name": "test",
+                        "state": "started",
+                        "slug": "test",
+                        "installed": True,
+                        "update_available": True,
+                        "version": "2.0.0",
+                        "version_latest": "2.0.1",
+                        "repository": "core",
+                        "url": "https://github.com/home-assistant/addons/test",
+                    },
+                    {
+                        "name": "test2",
+                        "state": "stopped",
+                        "slug": "test2",
+                        "installed": True,
+                        "update_available": False,
+                        "version": "3.1.0",
+                        "version_latest": "3.1.0",
+                        "repository": "core",
+                        "url": "https://github.com",
+                    },
+                ],
+            },
+        },
+    )
+    aioclient_mock.get(
+        "http://127.0.0.1/addons/test/stats",
+        json={
+            "result": "ok",
+            "data": {
+                "cpu_percent": 0.99,
+                "memory_usage": 182611968,
+                "memory_limit": 3977146368,
+                "memory_percent": 4.59,
+                "network_rx": 362570232,
+                "network_tx": 82374138,
+                "blk_read": 46010945536,
+                "blk_write": 15051526144,
+            },
+        },
+    )
+    aioclient_mock.get(
+        "http://127.0.0.1/ingress/panels", json={"result": "ok", "data": {"panels": {}}}
+    )
+
+
+@pytest.mark.parametrize(
+    "entity_id,expected",
+    [
+        ("binary_sensor.home_assistant_operating_system_update_available", "off"),
+        ("binary_sensor.test_update_available", "on"),
+        ("binary_sensor.test2_update_available", "off"),
+        ("binary_sensor.test_running", "on"),
+        ("binary_sensor.test2_running", "off"),
+    ],
+)
+async def test_binary_sensor(hass, entity_id, expected, aioclient_mock):
+    """Test hassio OS and addons binary sensor."""
+
+    with patch.dict(os.environ, MOCK_ENVIRON):
+        result = await async_setup_component(
+            hass,
+            "hassio",
+            {"http": {"server_port": 9999, "server_host": "127.0.0.1"}, "hassio": {}},
+        )
+        assert result
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data={}, unique_id=DOMAIN)
+    config_entry.add_to_hass(hass)
+
+    await hass.async_block_till_done()
+
+    """Check that entities are disabled by default."""
```

Comment:
Multiline strings are not used for code comments.
```suggestion
    # Check that entities are disabled by default.
```

Reply:
Identical in existing sensor test, but ok.